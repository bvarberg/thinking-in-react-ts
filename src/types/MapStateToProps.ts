import { State } from "./State"

export type MapStateToProps<Props, PropsOuter = {}> = (
  state: State,
  propsOuter: PropsOuter,
) => Props
