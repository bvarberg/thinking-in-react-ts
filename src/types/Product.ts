/**
 * Product - as represented by the starting point API example
 */
export interface Product {
  category: string
  price: string
  stocked: boolean
  name: string
}
