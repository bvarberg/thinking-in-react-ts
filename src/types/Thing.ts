/**
 * Thing - useless example interface
 * ------------------------------------------------
 * Must have height and width properties defined
 */

export interface Thing {
  width: number
  height: number
}
