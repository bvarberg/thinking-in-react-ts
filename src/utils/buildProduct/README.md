# buildProduct

## Overview

This builder can be used to create an object that implements the Product interface. It is meant to be used for testing.
