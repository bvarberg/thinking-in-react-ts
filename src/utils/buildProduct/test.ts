import { buildProduct } from "."

describe("buildProduct", () => {
  it("builds a default product fixture", () => {
    const received = buildProduct()

    expect(received).toEqual({
      category: "Sporting Goods",
      price: "$49.99",
      stocked: true,
      name: "Football",
    })
  })
})
