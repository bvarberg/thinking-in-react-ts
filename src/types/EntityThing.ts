import { Entity } from "./Entity"
import { Thing } from "./Thing"

/**
 * EntityThing - Entity and Thing
 * ------------------------------------------------
 * Must have id, height, width properties defined
 */

export interface EntityThing extends Entity, Thing {}
