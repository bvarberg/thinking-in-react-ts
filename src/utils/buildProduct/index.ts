import { Product } from "../../types/Product"

type BuildProduct = () => Product

export const buildProduct: BuildProduct = () => {
  return {
    category: "Sporting Goods",
    name: "Football",
    price: "$49.99",
    stocked: true,
  }
}
