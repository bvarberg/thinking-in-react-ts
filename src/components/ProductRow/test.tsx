import { ProductRow } from "."
import React from "react"
import { shallowRender } from "../../utils/shallowRender"
import { Product } from "../../types/Product"
import { buildProduct } from "../../utils/buildProduct"

describe("ProductRow", () => {
  it("renders", () => {
    const product: Product = buildProduct()

    const received = shallowRender(<ProductRow product={product} />)

    expect(received).toMatchSnapshot()
  })
})
