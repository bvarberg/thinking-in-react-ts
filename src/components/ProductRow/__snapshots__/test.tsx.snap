// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow renders 1`] = `
<tr
  className="ProductRow"
>
  <td>
    Football
  </td>
  <td>
    $49.99
  </td>
</tr>
`;
